src/main/java/org/folio/rest/impl/ItemStorageAPI.java

```diff
@@ -143,22 +144,15 @@ public void deleteItemStorageItemsByItemId(
 
   private CompletableFuture<Item> setEffectiveCallNumber(Map<String, String> okapiHeaders, Context vertxContext, Item item) {
     CompletableFuture<Item> completableFuture = null;
-    EffectiveCallNumberComponents components = new EffectiveCallNumberComponents();
-    if (item.getItemLevelCallNumber() != null && !item.getItemLevelCallNumber().isEmpty()) {
-      components.setCallNumber(item.getItemLevelCallNumber());
-      item.setEffectiveCallNumberComponents(components);
+    if (shouldNotRetrieveHoldingsRecord(item)) {
+      item.setEffectiveCallNumberComponents(EffectiveCallNumberComponentsUtil.buildComponents(null, item));
       completableFuture = CompletableFuture.supplyAsync(() -> item);
```

Comment:
I know this line isn't part of this PR, but it seems odd here to be using a Supplier for an value we already have. There is an overhead to creating a lambda and using the forkjoin pool to asynchronously supply the value. I think it makes sense to simplify this to:
```suggestion
      completableFuture = CompletableFuture.completedFuture(item);
```